### DISCUSSION TITLE
Why pathname did not match?

### DISCUSSION THREAD

**Discussion by inoyakaigor** [pillarjs/path-to-regexp#238] [Q&A]
I using react-router@5 which under the hood uses path-to-regexp@1.8.0

I have path pattern `/:projectKey/stat/:statType/:entryId?/(launch)?/:sublaunchpath?`. It must match pathname `/my_project/stat/campaigns/123456/launch/links-rating` and it does only if the word `launch` is surrounded by parentheses: `/(launch)?/`

So my question is why the part `/stat/` does not need parentheses but the `/launch/` part does?

**Comment by blakeembrey** [comment]
I tested it on http://forbeslindesay.github.io/express-route-tester/ (and locally with 1.8.0) and it does match without the parentheses, can you share the path pattern you were trying to use?

Edit: Local results:

<img width="985" alt="image" src="https://user-images.githubusercontent.com/1088987/103329829-80c28d80-4a13-11eb-91b4-e0e6a2fd700d.png">

**Reply by inoyakaigor** [reply]
I must have phrased the question not clearly. My bad. I should have explicitly said that the part `/launch/` is optional. I see in your screenshot that you removed parentheses with question mark.

**Reply by blakeembrey** [reply]
Does that mean you are trying to write `/launch?/`? The question mark there doesn’t mean anything, it doesn’t automatically know that the part you want to make optional is the preceding `launch` word which is why you must wrap it in parentheses. FWIW, there’s no example/README documentation that says this is possible so let me know if there’s somewhere the documentation can clarify this.
